src/frontend/planner_test/tests/testdata/output/union.yaml

```diff
@@ -253,25 +253,27 @@
     create table t2 (a int, b numeric, c bigint);
     select a from ((select * from t1 limit 1) union (select * from t2 limit 1)) T;
   optimized_logical_plan_for_batch: |-
-    LogicalAgg { group_key: [t1.a], aggs: [] }
-    └─LogicalUnion { all: true }
-      ├─LogicalLimit { limit: 1, offset: 0 }
-      │ └─LogicalScan { table: t1, columns: [t1.a] }
-      └─LogicalLimit { limit: 1, offset: 0 }
-        └─LogicalScan { table: t2, columns: [t2.a] }
+    LogicalProject { exprs: [t1.a] }
+    └─LogicalAgg { group_key: [t1.a, t1.b, t1.c], aggs: [] }
+      └─LogicalUnion { all: true }
+        ├─LogicalLimit { limit: 1, offset: 0 }
+        │ └─LogicalScan { table: t1, columns: [t1.a, t1.b, t1.c] }
+        └─LogicalLimit { limit: 1, offset: 0 }
+          └─LogicalScan { table: t2, columns: [t2.a, t2.b, t2.c] }
   batch_plan: |-
     BatchExchange { order: [], dist: Single }
-    └─BatchHashAgg { group_key: [t1.a], aggs: [] }
-      └─BatchExchange { order: [], dist: HashShard(t1.a) }
-        └─BatchUnion { all: true }
-          ├─BatchLimit { limit: 1, offset: 0 }
-          │ └─BatchExchange { order: [], dist: Single }
-          │   └─BatchLimit { limit: 1, offset: 0 }
-          │     └─BatchScan { table: t1, columns: [t1.a], distribution: SomeShard }
-          └─BatchLimit { limit: 1, offset: 0 }
-            └─BatchExchange { order: [], dist: Single }
-              └─BatchLimit { limit: 1, offset: 0 }
-                └─BatchScan { table: t2, columns: [t2.a], distribution: SomeShard }
```

Comment:
The old plan seems incorrect?

Reply:
I think it's because I moved `column_pruning` to the place after applying `CONVERT_DISTINCT_AGG_FOR_...`.